### PR TITLE
feat!: throw StandardException with a stringified error message

### PR DESCRIPTION
**Pull request by Aman-Aalam**
# Fixes #

Send a StandardException whenever the API returns an "errors" array response.
Earlier if the $response ["ok"] was false, the SDK would send a DownForMaintenance() exception, without specifying the API error even if the API sent back error messages.
Now, whatever error responses API returns in `$response["errors"]` are string-ified and sent as a message in StandardException, so developer know what went wrong.

### Checklist

- [x] I acknowledge that all my contributions will be made under the project's license
- [x] I have made a material change to the repo (functionality, testing, spelling, grammar)
- [x] I have titled the PR appropriately
- [x] I have updated my branch with the main branch
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added the necessary documentation about the functionality in the appropriate .md file
- [ ] I have added inline documentation to the code I modified

If you have questions, create a GitHub Issue in this repository.

### CODE REVIEW

File: tests/Setup.php [hunk 1]
```diff
@@ -19,8 +20,8 @@ public static function doSetup()
     {
         Configuration::reset();
 
-        Configuration::environment('development');
-        Configuration::publicKey('integration_public_key');
-        Configuration::privateKey('integration_private_key');
+        Configuration::environment('production');
+        Configuration::publicKey('ASC7AsydNKAKBB5BNEVFVZ0P');
+        Configuration::privateKey('4vapebxbpq26gmk7a2zdj837u0qn24ufynqnnjfq');
```
Comment (Yongyiw) [inline tests/Setup.php]: @Sheikh-Aman Red flag for showing private key and public key in a public repo....

File: lib/PaymentRails/BalanceGateway.php [hunk 2]
```diff
@@ -56,6 +56,8 @@ public function search($query)
             }, $response['balances']);
 
             return new ResourceCollection($response, $items, $pager);
+        } else if ($response['errors']){
+            throw new Exception\Standard($response['errors']);
         } else {
             throw new Exception\DownForMaintenance();
```
Comment (Yongyiw) [inline lib/PaymentRails/BalanceGateway.php:62]: So DownForMaintenance() will never been throw - maybe HTTP 500 go to DownForMaintenance but 400 show the concat error strign.

File: tests/Setup.php (hunk 1 above)
Comment (Aman-Aalam) [reply tests/Setup.php]: Thanks @Yongyiw , fixed this.

File: lib/PaymentRails/BalanceGateway.php (hunk 2 above)
Comment (Yongyiw) [reply lib/PaymentRails/BalanceGateway.php:62]: also maybe have a handleBadRequest in base Gateway/Http class.. 

and put the error response handler there and add `extends BaseGateway` for each Gateway. that way maybe we don't need to update 10 files..

Comment (Aman-Aalam) [reply lib/PaymentRails/BalanceGateway.php:62]: Yes.
As long as the response has a non-empty "errors" array, all the error messages will be concatenated into one string and raised as a StandardException

StandardExcetion already existing, so I used it.
We could make another class of our own if that suits.

**Review by Yongyiw — APPROVED** [review]

**Review by koblas — APPROVED** [review]

**Review by fronugget — APPROVED** [review]

**Pull request merged**